scripts/include/hook-common

```diff
@@ -0,0 +1,42 @@
+default_layouts="us"
+layouts_file="$SNAP/etc/layouts.txt"
+osk_layout_file="$SNAP_DATA/osk-layout.txt"
+
+# Use the first argument to join all the remaining ones
+function join {
+  local IFS="$1"
+  shift
+  echo "$*"
+}
+
+# Loop through list of layouts ensuring entries are valid and comma separated
+# Prints all the valid layouts and returns an error if any was not.
+function validate_layouts {
+  local error=0
+  for layout in $(snapctl get layout | tr ',' ' '); do
+    if grep -Fxq "$layout" "$layouts_file"; then
+      echo ${layout}
+    else
+      echo "Error: $layout is not a valid layout ID." >&2
+      echo "       See $SNAP_NAME.list-layouts for a list of valid layouts." >&2
+      error=1
+    fi
+  done
+  return $error
+}
+
+# Prints "yes" or "no" depending on whether the file changed or not
+function set_layouts {
+  local osk_layouts
+  if [ "$#" -gt 0 ]; then
+    osk_layouts="$(join , "${@}")"
+  else
+    osk_layouts="${default_layouts}"
+  fi
+  snapctl set "layout=${osk_layouts}"
+  if [ "$(cat "${osk_layout_file}")" != "${osk_layouts}" ]; then
+    echo "${osk_layouts}" > "${osk_layout_file}"
+    echo yes
+  fi
+  echo no
```

Comment:
It looks to me like this will print `no` always, even after printing `yes`

Reply:
Indeed. Fixed.